### DISCUSSION TITLE
TLS Certificate Request Error

### DISCUSSION THREAD

**Discussion by kurtbruns** [community/community#23372] [Other Feature Feedback, Questions, & Ideas]
<p>Hi, I just switched my workflow to use Github Actions to deploy my website: <a href="http://wumbo.net" rel="noopener nofollow ugc">wumbo.net</a>, however when renaming the Github pages branch the settings &gt; pages page shows that the TLS certificate is stuck showing an error.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/a/8/a8dddda007980a02ea7b3dc5d86ddd20ebbe607a.png" data-download-href="/uploads/short-url/o5RyJeBvA64rmoCTWhbvuROTjXs.png?dl=1" title="Screen Shot 2021-07-16 at 3.39.56 PM" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/12160501/181076606-e2559645-4abc-4609-aaf3-1fc55b5b98e7.png" alt="Screen Shot 2021-07-16 at 3.39.56 PM" data-base62-sha1="o5RyJeBvA64rmoCTWhbvuROTjXs" width="690" height="187" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/a/8/a8dddda007980a02ea7b3dc5d86ddd20ebbe607a_2_690x187.png, https://aws1.discourse-cdn.com/github/optimized/3X/a/8/a8dddda007980a02ea7b3dc5d86ddd20ebbe607a_2_1035x280.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/a/8/a8dddda007980a02ea7b3dc5d86ddd20ebbe607a_2_1380x374.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/a/8/a8dddda007980a02ea7b3dc5d86ddd20ebbe607a_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2021-07-16 at 3.39.56 PM</span><span class="informations">1878×510 62.7 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>It looks like the old certificate is still valid when using HTTPS, but I really would like to enforce HTTPS only traffic. Is there anything that I can do other than waiting?</p>
<p>Edit: Since my DNS was working right before changing my workflow and HTTPS was enforced, I am assuming that DNS is still configured correctly.</p>

**Comment by kurtbruns** [comment]
<p>I used the <a href="https://github.com/github/pages-health-check" rel="noopener nofollow ugc">pages health check tool</a> today to see if I could get any more information on the TLS Certificate Error. Here is the output:</p>
<pre><code class="lang-auto">{:host=&gt;"wumbo.net",
 :uri=&gt;"http://wumbo.net/",
 :nameservers=&gt;:default,
 :dns_resolves?=&gt;true,
 :proxied?=&gt;false,
 :cloudflare_ip?=&gt;false,
 :fastly_ip?=&gt;false,
 :old_ip_address?=&gt;false,
 :a_record?=&gt;true,
 :cname_record?=&gt;false,
 :mx_records_present?=&gt;true,
 :valid_domain?=&gt;true,
 :apex_domain?=&gt;true,
 :should_be_a_record?=&gt;true,
 :cname_to_github_user_domain?=&gt;false,
 :cname_to_pages_dot_github_dot_com?=&gt;false,
 :cname_to_fastly?=&gt;false,
 :pointed_to_github_pages_ip?=&gt;true,
 :non_github_pages_ip_present?=&gt;false,
 :pages_domain?=&gt;false,
 :served_by_pages?=&gt;true,
 :valid?=&gt;true,
 :reason=&gt;nil,
 :https?=&gt;true,
 :enforces_https?=&gt;false,
 :https_error=&gt;nil,
 :https_eligible?=&gt;true,
 :caa_error=&gt;nil,
 :dns_zone_soa?=&gt;true,
 :dns_zone_ns?=&gt;true}
</code></pre>
<p>Currently, in my DNS (Google Domains) configuration I am forwarding the subdomain <code>www</code> traffic to the apex domain. See attached screenshot. The docs say that it is <em>recommended</em> to create a CNAME record for for the <code>www</code> subdomain, but not <em>required</em> – so I am assuming <code> :cname_record?=&gt;false,</code> isn’t the source of the error.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/0/2/02a68e4ca52e84eac608a2fe08fb29c1ed71eba3.png" data-download-href="/uploads/short-url/nrNycdy1Ha3YVJE7yWBbvXcZVN.png?dl=1" title="Screen Shot 2021-07-17 at 8.28.23 AM" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/12160501/181076607-d584b3a7-aca8-4fb0-9030-183d59aaf657.png" alt="Screen Shot 2021-07-17 at 8.28.23 AM" data-base62-sha1="nrNycdy1Ha3YVJE7yWBbvXcZVN" width="690" height="57" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/0/2/02a68e4ca52e84eac608a2fe08fb29c1ed71eba3_2_690x57.png, https://aws1.discourse-cdn.com/github/optimized/3X/0/2/02a68e4ca52e84eac608a2fe08fb29c1ed71eba3_2_1035x85.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/0/2/02a68e4ca52e84eac608a2fe08fb29c1ed71eba3_2_1380x114.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/0/2/02a68e4ca52e84eac608a2fe08fb29c1ed71eba3_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2021-07-17 at 8.28.23 AM</span><span class="informations">1966×164 16.8 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by kurtbruns** [comment, marked as answer]
<p>Update for those in a similar situation. Adding a <code>www</code> CNAME record fixed the error following the advice from <a href="https://github.community/t/certificate-request-error-is-persistent-tls-certificate-cant-be-provisioned/11008">this post</a>. My DNS custom resource records section now looks like this:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/e/a/eaed54a9a50374cf8af00326f6996a41a13ef507.png" data-download-href="/uploads/short-url/xwgcOUL2VVeLMibkwg8DLOhh5Ph.png?dl=1" title="Screen Shot 2021-07-18 at 10.55.08 AM" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/12160501/181076609-b4efbbd5-355d-4619-acc6-eb646a3e33fc.png" alt="Screen Shot 2021-07-18 at 10.55.08 AM" data-base62-sha1="xwgcOUL2VVeLMibkwg8DLOhh5Ph" width="690" height="330" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/e/a/eaed54a9a50374cf8af00326f6996a41a13ef507_2_690x330.png, https://aws1.discourse-cdn.com/github/optimized/3X/e/a/eaed54a9a50374cf8af00326f6996a41a13ef507_2_1035x495.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/e/a/eaed54a9a50374cf8af00326f6996a41a13ef507_2_1380x660.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/e/a/eaed54a9a50374cf8af00326f6996a41a13ef507_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2021-07-18 at 10.55.08 AM</span><span class="informations">2070×990 140 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>You can trigger the TLS certificate process by temporarily removing your custom domain and adding it again. A couple minutes later I was able to check the “Enforce HTTPS” check box again.</p>

**Reply by kananinirav** [reply]
Thank you @kurtbruns

**Comment by SwampDonkeyDW** [comment]
So, I had a working portfolio site with a TLS cert before. However, I decided to create a new site using React/Next.js. I got everything up and running with my main and gh-pages branches merged. However, now I cannot seem to reuse my domain. The DNS check passes, but it errors when I am awaiting the TLS certificate. It has been well over 24 hours now and still nothing. Everything was setup correctly previously, so I don't know what's up this time around.

**Reply by notsag-dev** [reply]
@travisengh Same issue here. Reusing the domain make the certificate provision get stuck in that error state. Did it work out for you in the end?

**Reply by tronch0** [reply]
@travisengh  @notsag-dev Any update regarding this? Im on the same boat :/

**Reply by SwampDonkeyDW** [reply]
So, what I ended up doing was deleting that repository, waiting about two days, and then creating a completely new repository to store my website on. It still took awhile, but it worked.

I would also suggest self hosting your site through a cloudflare tunnel if you can. I recently containerized my website and did this on a raspberry pi docker swarm and will never look back. :)

**Reply by notsag-dev** [reply]
@tronch0 it eventually worked for me. I believe I had to wait for more than an entire day, but I didn't recreate the repo. For the record, did it work for you already?

**Reply by tronch0** [reply]
@travisengh @notsag-dev thanks for your inputs.

I waiting for about 48h but still got the cert error, so I ended up changing the address I deployed to, instead of the regular address I used a subdomain to have a new address for the ssl certificate.

*my project is a personal website so I was more lean with my address constraints.

Thanks again! :)

**Reply by SwampDonkeyDW** [reply]
I had waited over a week before I deleted and recreated the repository. It just wasn't working for me until I redid it all. Mine was for a portfolio website, so I wasn't in much of a rush.

**Comment by AsifaJahanSafa** [comment]
What should i do to solver this problem in my windows 7 32bit laptop "The request was aborted: Could not create SSL/TLS secure channal"
